pkg/kube/client.go

```diff
@@ -105,6 +109,64 @@ func (c *Client) Create(resources ResourceList) (*Result, error) {
 	return &Result{Created: resources}, nil
 }
 
+func transformRequests(req *rest.Request) {
+	group := metav1beta1.GroupName
+	version := metav1beta1.SchemeGroupVersion.Version
+
+	tableParam := fmt.Sprintf("application/json;as=Table;v=%s;g=%s, application/json", version, group)
```

Comment:
This should also accept the v1 as well as the v1beta1. You can see the work I did for Helm v2 on this at #7785.

Reply:
Hi Matt,

   Thanks for the explanation. As you suggest, i made changes inorder to support both metav1 and metav1beta1 cases. I had gone through the above PR link and used the structured metadata for JSON/YAML. Everything is working fine.

Below are the code-snippet changes.

```
func transformRequests(req *rest.Request) {
        tableParam := strings.Join([]string{
                fmt.Sprintf("application/json;as=Table;v=%s;g=%s", metav1.SchemeGroupVersion.Version, metav1.GroupName),
                fmt.Sprintf("application/json;as=Table;v=%s;g=%s", metav1beta1.SchemeGroupVersion.Version, metav1beta1.GroupName),
                "application/json",
        }, ",")
        req.SetHeader("Accept", tableParam)

        // if sorting, ensure we receive the full object in order to introspect its fields via jsonpath
        req.Param("includeObject", "Object")
}

func sortTableSlice(objs []runtime.Object) []runtime.Object {
        if len(objs) < 2 {
                return objs
        }

        ntbl := &metav1.Table{}
        // Sort the list of objects
        var newObjs []runtime.Object
        namesCache := make(map[string]runtime.Object, len(objs))
        var names []string
        for _, obj := range objs {
                unstr, ok := obj.(*unstructured.Unstructured)
                if !ok {
                        return objs
                }
                if err := runtime.DefaultUnstructuredConverter.FromUnstructured(unstr.Object, ntbl); err != nil {
                        return objs
                }
                namesCache[ntbl.GetSelfLink()] = obj
                names = append(names, ntbl.GetSelfLink())
        }
        sort.Strings(names)
        for _, name := range names {
                newObjs = append(newObjs, namesCache[name])
        }
        return newObjs
}


func (c *Client) Get(resources ResourceList) (string, error) {
        buf := new(bytes.Buffer)
        printFlags := get.NewHumanPrintFlags()
        typePrinter, _ := printFlags.ToPrinter("")
        printer := &get.TablePrinter{Delegate: typePrinter}
        objs := make(map[string][]runtime.Object)
        err := resources.Visit(func(info *resource.Info, err error) error {
                if err != nil {
                        return err
                }
                gvk := info.ResourceMapping().GroupVersionKind
                vk := gvk.Version + "/" + gvk.Kind
                obj, err := getResource(info)
                if err != nil {
                        return err
                }
                objs[vk] = append(objs[vk], obj)
                return nil
        })
        if err != nil {
                return "", err
        }
        var sortingkeys []string
        for key := range objs {
                sortingkeys = append(sortingkeys, key)
        }
        // Sort alphabetically by version/kind keys
        sort.Strings(sortingkeys)
        for _, t := range sortingkeys {
                if _, err = fmt.Fprintf(buf, "==> %s\n", t); err != nil {
                        return "", err
                }
                vk := objs[t]
                vk = sortTableSlice(vk)
                for _, resource := range vk {
                        if err := printer.PrintObj(resource, buf); err != nil {
                                c.Log("failed to print object type %s: %v", t, err)
                                return "", err
                        }
                }
                if _, err := buf.WriteString("\n"); err != nil {
                        return "", err
                }
        }
        return buf.String(), nil
}

func getResource(info *resource.Info) (runtime.Object, error) {
        obj, err := resource.NewHelper(info.Client, info.Mapping).Get(info.Namespace, info.Name, false)
        if err != nil {
                return nil, err
        }
        return obj, nil
}
```

Could you please look into it and let me know is this okay or not. If yes, then i will commit the code changes.

Reply:
This part looks ok. Once I have the PR to review in a full code editor I can do a better review. I look forward to seeing your changes.

Reply:
Hi Matt,

   I had commited the changes. Could you please review the code.